From: Christoph Hellwig <hch@infradead.org>
Date: Tue, 24 Apr 2012 00:25:09 -0400
Subject: [PATCH] target: remove the t_se_count field in struct se_cmd
Git-commit: d5dc28eb92f2a2305a02cb3a5f1ed36542d47512
References: FATE#313550
Patch-Mainline: v3.5

Now that tasks are gone we are guaranteed to only get a single completion
per command, and thus don't need this counter.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/target/target_core_transport.c |    6 ------
 include/target/target_core_base.h      |    1 -
 2 files changed, 0 insertions(+), 7 deletions(-)

diff --git a/drivers/target/target_core_transport.c b/drivers/target/target_core_transport.c
index cd77f87..24ca652 100644
--- a/drivers/target/target_core_transport.c
+++ b/drivers/target/target_core_transport.c
@@ -3289,11 +3289,6 @@ static void transport_put_cmd(struct se_cmd *cmd)
 			goto out_busy;
 	}
 
-	if (atomic_read(&cmd->t_se_count)) {
-		if (!atomic_dec_and_test(&cmd->t_se_count))
-			goto out_busy;
-	}
-
 	if (cmd->transport_state & CMD_T_DEV_ACTIVE) {
 		cmd->transport_state &= ~CMD_T_DEV_ACTIVE;
 		target_remove_from_state_list(cmd);
@@ -3503,7 +3498,6 @@ int transport_generic_new_cmd(struct se_cmd *cmd)
 	}
 
 	atomic_inc(&cmd->t_fe_count);
-	atomic_inc(&cmd->t_se_count);
 
 	/*
 	 * For WRITEs, let the fabric know its buffer is ready.
diff --git a/include/target/target_core_base.h b/include/target/target_core_base.h
index 6c83b14..927771a 100644
--- a/include/target/target_core_base.h
+++ b/include/target/target_core_base.h
@@ -548,7 +548,6 @@ struct se_cmd {
 	unsigned char		__t_task_cdb[TCM_MAX_COMMAND_SIZE];
 	unsigned long long	t_task_lba;
 	atomic_t		t_fe_count;
-	atomic_t		t_se_count;
 	unsigned int		transport_state;
 #define CMD_T_ABORTED		(1 << 0)
 #define CMD_T_ACTIVE		(1 << 1)
-- 
1.7.4.2

